src/rules/tryceratops/rules/raise_within_try.rs

```diff
@@ -0,0 +1,53 @@
+use ruff_macros::derive_message_formats;
+use rustpython_ast::{Stmt, StmtKind};
+
+use crate::ast::types::Range;
+use crate::ast::visitor::{self, Visitor};
+use crate::checkers::ast::Checker;
+use crate::define_violation;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+define_violation!(
+    pub struct RaiseWithinTry;
+);
+impl Violation for RaiseWithinTry {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Abstract `raise` to an inner function")
+    }
+}
+
+#[derive(Default)]
+struct RaiseStatementVisitor<'a> {
+    raises: Vec<&'a Stmt>,
+}
+
+impl<'a, 'b> Visitor<'b> for RaiseStatementVisitor<'a>
+where
+    'b: 'a,
+{
+    fn visit_stmt(&mut self, stmt: &'b Stmt) {
+        match stmt.node {
+            StmtKind::Raise { .. } => self.raises.push(stmt),
+            _ => visitor::walk_stmt(self, stmt),
```

Comment:
Should we avoid recursing if we hit another `try` block?

Reply:
Otherwise, we might raise the same error multiple times for nested `try` blocks.

Reply:
Done.